avm/res/hybrid-compute/machine/main.bicep

```
@@ -0,0 +1,256 @@
+metadata name = 'Hybrid Compute Machines'
+metadata description = 'This module deploys an Arc Machine for use with Arc Resource Bridge for Azure Stack HCI or VMware. In these scenarios, this resource module will be used in combination with another resource module to create the require Virtual Machine Instance extension resource on this Arc Machine resource. This module should not be used for other Arc-enabled server scenarios, where the Arc Machine resource is created automatically by the onboarding process.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. The name of the Arc machine to be created. You should use a unique prefix to reduce name collisions in Active Directory.')
+param name string
+
+@description('Required. Kind of Arc machine to be created. Possible values are: HCI, SCVMM, VMware.')
+param kind string
+
+@description('Conditional. The resource ID of an Arc Private Link Scope which which to associate this machine. Required if you are using Private Link for Arc and your Arc Machine will resolve a Private Endpoint for connectivity to Azure.')
+param privateLinkScopeResourceId string = ''
+
+@description('Optional. Parent cluster resource ID (Azure Stack HCI).')
+param parentClusterResourceId string = ''
+
+@description('Optional. The GUID of the on-premises virtual machine from your hypervisor.')
+param vmId string = ''
+
+@description('Optional. The Public Key that the client provides to be used during initial resource onboarding.')
+param clientPublicKey string = ''
```

```suggestion
@secure()
param clientPublicKey string = ''
```

I know, I know, it's just a public key. But I know that some folks are very particular about also treating this as a credential, so here we go.

Please note, once updated, requires also a re-run of the `Set-AVMModule` script on the module folder